pgserver/connection_handler.go

```diff
@@ -663,39 +897,52 @@ func (h *ConnectionHandler) handleParse(message *pgproto3.Parse) error {
 		return nil
 	}
 
-	stmt, params, fields, err := h.duckHandler.ComPrepareParsed(context.Background(), h.mysqlConn, query.String, query.AST)
-	if err != nil {
-		return err
-	}
+	handledOutsideEngine := shouldQueryBeHandledInPlace(query.String)
 
-	if !query.PgParsable {
-		query.StatementTag = GetStatementTag(stmt)
-	}
+	if !handledOutsideEngine {
```

Comment:
Please keep the original control flow as unchanged as possible. Special cases should be handled in a positive `if` block, i.e., `if handleOutsideEngine`, rather than modifying the original logic to be in a `if !handledOutsideEngine` block.

Reply:
In addition, why we need the `handledOutsideEngine` flag? Isn't `Stmt == nil` enough?

Reply:
fixed